controllers/watcherapi_controller.go

```diff
@@ -232,7 +232,13 @@ func (r *WatcherAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		// but we have to return while waiting for the service to be exposed
 		return ctrl.Result{}, err
 	}
-	_ = apiEndpoints // we'll use the apiEndpoints later
+
+	result, err = r.ensureKeystoneEndpoint(ctx, helper, instance, apiEndpoints)
+	if (err != nil || result != ctrl.Result{}) {
+		// We can ignore RequeueAfter as we are watching the KeystoneEndpoint
+		// resource
+		return ctrl.Result{}, err
```

Comment:
```suggestion
		return result, err
```

Reply:
fixed, will push the fix in the next update